app/scripts/lib/typed-message-manager.js

```diff
@@ -204,7 +204,7 @@ export default class TypedMessageManager extends EventEmitter {
             `Cannot sign messages for chainId "${chainId}", because MetaMask is switching networks.`,
           );
           if (typeof chainId === 'string') {
-            chainId = parseInt(chainId, 16);
+            chainId = parseInt(chainId, chainId.startsWith('0x') ? 16 : 10);
```

Comment:
I've tested the fix at https://github.com/MetaMask/metamask-extension/tree/revert-signtypeddata-validation which fixes this issue without the parsing issues mentioned in this thread. :)